app/src/main/java/org/openobservatory/ooniprobe/domain/MeasurementsManager.java

```diff
@@ -109,6 +117,57 @@ public String getReadableEntry(Measurement measurement) throws IOException {
         return jsonPrinter.prettyText(FileUtils.readFileToString(entryFile, StandardCharsets.UTF_8));
     }
 
+    public void syncMeasurements(Activity activity,
+                                 OnStartResubmission onStartResubmission,
+                                 OnCancelResubmission onCancelResubmission,
+                                 Integer... params) {
+
+        if (determineIfMeasurementsWereTakenOverVPN(params)) {
+            new AlertDialog.Builder(activity, R.style.MaterialAlertDialogCustom)
+                    .setTitle(activity.getString(R.string.Modal_UploadVPNResults_Title))
+                    .setMessage(activity.getString(R.string.Modal_UploadVPNResults_Message))
+                    .setNegativeButton(R.string.Modal_Cancel, (dialogInterface, i) -> {
+                        dialogInterface.dismiss();
+                        onCancelResubmission.onCancel();
+                    })
+                    .setPositiveButton(R.string.Modal_OK, (dialogInterface, i) -> {
+                        onStartResubmission.onResubmit();
+                    })
+                    .show();
```

Comment:
Is there a way to factor this piece of code which seems repeated ~equally more than once?

Reply:
@bassosimone this is a normal alert, where else is it repeated?